spec/classes/install_spec.rb

```diff
@@ -275,4 +283,53 @@
     end
   end
 
-end
\ No newline at end of file
+  context 'installing from tar archive' do
+    let(:user) { 'zookeeper' }
+    let(:group) { 'zookeeper' }
+    let(:package_dir) { '/tmp/zookeeper' }
+    let(:install_dir) { '/opt/zookeeper' }
+    let(:vers) { '3.4.5' }
+    let(:mirror_url) { 'http://mirror.cogentco.com/pub/apache' }
+    let(:basefilename) { "zookeeper-#{vers}.tar.gz" }
+    let(:package_url) { "#{mirror_url}/zookeeper/zookeeper-#{vers}/zookeeper-#{vers}.tar.gz" }
+    let(:extract_path) { "#{install_dir}-#{vers}" }
+
+    let(:facts) {{
+      :operatingsystem => 'Ubuntu',
+      :osfamily => 'Debian',
+      :lsbdistcodename => 'trusty',
+    }}
+
+    let(:params) { {
+      :install_method => 'tar',
+      :package_dir    => package_dir,
+      :install_dir    => install_dir,
+      :ensure         => vers,
+      :mirror_url     => mirror_url,
+    } }
+
+    it { should contain_file(package_dir).with({
+      :ensure => 'directory',
+      :owner  => user,
+      :group  => group,
+    })}
+    it { should contain_file(extract_path).with({
+      :ensure => 'directory',
+      :owner  => user,
+      :group  => group,
+    })}
+    it { should contain_file(install_dir).with({
+      :ensure => 'link',
+      :target => extract_path,
+    })}
+    it { should contain_archive("#{package_dir}/#{basefilename}").with({
```

Comment:
Grr, not a fan of multiline blocks using `{`. Prefer `it do` for multiline blocks.